src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DoctrineQueryExtensionPass.php

```diff
@@ -54,6 +54,12 @@ private function findSortedServices(ContainerBuilder $container, $tag)
     {
         $extensions = [];
         foreach ($container->findTaggedServiceIds($tag) as $serviceId => $tags) {
+            if ('api_platform.doctrine.orm.query_extension.eager_loading' === $serviceId
```

Comment:
IMO it should be done in the `ApiPlatformExtension` class with the `ContainerBuilder::removeDefinition()` method.